packages/story-editor/src/components/library/panes/shopping/productList.js

```diff
@@ -0,0 +1,224 @@
+  height: 16px;
+  border-radius: 100%;
+  color: ${({ theme }) => theme.colors.interactiveBg.disable};
+`;
+
+const StyledActionsContainer = styled.div`
+  margin-right: 10px;
+
+  .check {
+    ${baseIcon}
+    background-color: ${({ theme }) => theme.colors.fg.positive};
+  }
+
+  .remove {
+    ${baseIcon}
+    background-color: ${({ theme }) => theme.colors.fg.negative};
+    display: none;
+  }
+
+  button {
+    margin-right: 10px;
+    &:focus,
+    &:hover {
+      .remove {
+        display: block;
+      }
+      .check {
+        display: none;
+      }
+    }
+  }
+`;
+
+const imgPlaceholder = css`
+  display: block;
+  margin-right: 10px;
+  border-radius: 4px;
+  background-color: ${({ theme }) => theme.colors.interactiveBg.disable};
+  width: 41px;
+  height: 41px;
+`;
+
+const StyledImgPlaceHolder = styled.div`
+  ${imgPlaceholder}
+`;
+
+const StyledImage = styled.img`
+  ${imgPlaceholder}
+`;
+
+const StyledDescription = styled.div`
+  padding-left: 12px;
+`;
+
+const StyledTitle = styled(Text).attrs({
+  size: THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.MEDIUM,
+})`
+  color: ${({ theme }) => theme.colors.fg.primary};
+`;
+
+const StyledPrice = styled(Text)`
+  color: ${({ theme }) => theme.colors.fg.tertiary};
+`;
+
+const onPageIcons = (
+  <>
+    <Icons.Checkmark className="check" /> <Icons.Cross className="remove" />
+  </>
+);
+function ProductList({
+  products = [],
+  onPageProducts = [],
+  onInsert,
+  onRemove,
+  isMenuFocused = true,
+}) {
+  const { handleListNav, currentRowsRef, setCurrentFocusIndex } =
+    useProductNavigation({ isMenuFocused, products });
+
+  useEffect(() => {
+    if (!isMenuFocused) {
+      setCurrentFocusIndex(0);
+    }
+  }, [isMenuFocused, setCurrentFocusIndex]);
+
+  return (
+    // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions -- list handles arrow up and arrow down
+    <div role="list" onKeyDown={handleListNav}>
+      {products.map((product, index) => {
+        const imageSrc = product?.productImages[0]?.url || '';
+        const onPage = isOnPage(product, onPageProducts);
+        const handleClick =
+          onPage && onPage?.productId
+            ? () => {
+                onRemove && onRemove(product);
+              }
+            : () => {
+                onInsert && onInsert(product);
+              };
+
+        const handleFocus = (evt) => {
+          if (currentRowsRef.current[`row-${index}`] && evt.type === 'focus') {
+            setCurrentFocusIndex(index);
+          }
+        };
+
+        const ADD_PRODUCT_TEXT = sprintf(
+          /* translators: %s: add product title.*/
+          __('Add %s', 'web-stories'),
+          product.productTitle
+        );
+
+        const REMOVE_PRODUCT_TEXT = sprintf(
+          /* translators: %s: remove product title.*/
+          __('Remove %s', 'web-stories'),
+          product.productTitle
+        );
+
+        return (
+          <StyledListItem
+            id={`row-${index}`}
```

Comment:
Better to use the product ID here instead of the index

Reply:
Will remove --- some of this code was pulled over from the custom fonts listbox .... in this case we don't really need it.